shared/engine/index.js

```diff
@@ -1,430 +1,284 @@
 // @flow
```

Comment:
tldr; Before the engine had maps to keep track of session ids, responses, outstanding requests, etc. Now it just knows about sessions and generalized handlers mostly